### PR TITLE
SALTO-873 Multienv - suggest to isolate the existing env when adding a 2nd env

### PR DESCRIPTION
**Pull request by netama**
* Add a prompt when adding the 2nd env if some configuration exists and no configuration is isolated (if something is in the env folder we assume the user made a conscious decision about what should remain in the common folder)
* Add a `force` flag to the env command to avoid this interactive prompt

### CODE REVIEW

File: packages/cli/e2e_test/multi_env.test.ts [hunk 1]
```diff
@@ -552,6 +552,19 @@ describe('multi env tests', () => {
           expect(obj.annotationTypes.metadataType).toBeDefined()
         }))
       })
+
+      it('should not update common with the attributes added in the deploy', async () => {
```
Comment (netama) [inline packages/cli/e2e_test/multi_env.test.ts:556]: not related to this PR - replaces [this change](https://github.com/salto-io/salto/pull/1269/files#diff-c90bec8f7ee92483f615e645ca8da141L543)

File: packages/cli/src/prompts.ts [hunk 2]
```diff
@@ -159,26 +159,15 @@ ${Prompts.SERVICE_ADD_HELP}`
   public static readonly RENAME_ENV = (currentEnvName: string, newEnvName: string): string =>
     `Renamed environment - ${currentEnvName} -> ${newEnvName}`
 
-  public static readonly ISOLATED_MODE_FOR_NEW_ENV_RECOMMENDATION = 'The current fetch command is running for the first time for this environment.'
```
Comment (netama) [inline packages/cli/src/prompts.ts:162]: these are no longer used (leftovers from a previous PR)

File: packages/cli/src/commands/env.ts [hunk 3]
```diff
@@ -155,9 +197,18 @@ const envsBuilder = createCommandBuilder({
         desc: 'The new name of the environment (required for rename)',
       },
     },
+    keyed: {
+      force: {
```
Comment (ori-moisis) [inline packages/cli/src/commands/env.ts:201]: I admit I don't fully understand the flow and all the new commands, but generally with CLI interfaces if there is a "force" flag that makes the command non-interactive, the different possible interactive choices should remain possible in non-interactive flow (in this case, if it was possible in the interactive flow to either move the elements or not move the elements, that choice should still be possible in the non-interactive flow)

Comment (netama) [reply packages/cli/src/commands/env.ts:201]: I agree in general, and wasn't sure what would be the best option here - ended up following the pattern from the workspace warnings before fetch (that are ignored with `--force`), and from the pretty similar `fetch --isolate` that existed until recently (you can see it [here](https://github.com/roironn/salto-1/blob/6f6cde8de28ec3bd9a1f5c2f49561c1f4b637545/packages/cli/src/commands/fetch.ts#L125) ). 

But I'm happy to add that as a direct interaction too - any suggestions for a flag / enum to control that?

File: packages/cli/src/prompts.ts [hunk 4]
```diff
@@ -159,26 +159,15 @@ ${Prompts.SERVICE_ADD_HELP}`
...
-    oldServicesNames: string
-  ): string => `The current fetch command is running for the first time for ${newServicesNames} in isolated mode.`
-    + ` This will also fetch ${oldServicesNames} in strict mode and may result in unwanted changes to `
-    + ' other environments.\n'
-    + `It is recommended to run this fetch as an isolated fetch only for ${newServicesNames}`
-
-  public static readonly APPROVE_ISOLATED_RECOMMENDATION = (
-    newServicesNames: string
-  ): string => `Would you like to switch to isolated mode and fetch ${newServicesNames}? (Answer No to continue without switching)`
+  public static readonly ISOLATE_FIRST_ENV_RECOMMENDATION = (
```
Comment (adamro) [inline packages/cli/src/prompts.ts:162]: The user didn't know the concept of envs at all till now and doesn't know the concept of common at all. Might be a good place to give a short explanation.

I am not good enough @ copy to even give an example :) 

But something about - 
1. This action moving this workspace from being a single env to multi-env workspace
2. Introducing the concept of common and env specific
only then
3. Recommend that the default env move to env specific

File: packages/cli/src/prompts.ts [hunk 5]
```diff
@@ -159,26 +159,15 @@ ${Prompts.SERVICE_ADD_HELP}`
...
+    existingEnv: string
+  ): string => 'This action will add a second environment to the workspace.'
+    + ` It is recommended to move environment ${existingEnv} out of the common configuration before proceeding.`
+
+  public static readonly APPROVE_ISOLATE_BEFORE_MULTIENV_RECOMMENDATION = (
+    existingEnv: string
+  ): string => `Would you like to isolate the configuration for ${existingEnv} before adding the new environment?`
```
Comment (adamro) [inline packages/cli/src/prompts.ts:169]: The command to move configuration from common to env specific is called "demote" atm, so using the term isolate is a bit out of place, no? (I think we need this to be aligned with the copy of the above message anyway which I commented on).

All of this can go in a different PR if you want btw

File: packages/cli/src/commands/env.ts [hunk 6]
```diff
@@ -35,11 +39,47 @@ const setEnvironment = async (
...
+): Promise<boolean> => {
+  const envNames = workspace.envs()
+  return (
+    envNames.length === 1
+    && !await workspace.isEmpty(true)
+    && !await envFolderExists(workspaceDir, envNames[0])
+  )
+}
+
+const isolateExistingEnvironment = async (workspace: Workspace): Promise<void> => {
```
Comment (adamro) [inline packages/cli/src/commands/env.ts:54]: `isolateExistingEnvironments` (this is what it actually does even if it is used for one :) )

File: packages/cli/src/commands/env.ts [hunk 7]
```diff
@@ -35,11 +39,47 @@ const setEnvironment = async (
...
+  output: CliOutput,
+  workspace: Workspace,
+  workspaceDir: string,
+  force?: boolean,
+): Promise<void> => {
+  if (!force && await shouldRecommendToIsolateCurrentEnv(workspace, workspaceDir)) {
+    const existingEnv = workspace.envs()[0]
+    outputLine(output, formatApproveIsolatCurrentEnvPrompt(existingEnv))
+    if (await cliApproveIsolateBeforeMultiEnv(existingEnv)) {
+      await isolateExistingEnvironment(workspace)
```
Comment (adamro) [inline packages/cli/src/commands/env.ts:69]: I think it makes sense to print about finishing to demote the current env here

File: packages/cli/src/commands/env.ts [hunk 8]
```diff
@@ -35,11 +39,47 @@ const setEnvironment = async (
...
 
+const shouldRecommendToIsolateCurrentEnv = async (
+  workspace: Workspace,
+  workspaceDir: string,
+): Promise<boolean> => {
+  const envNames = workspace.envs()
+  return (
+    envNames.length === 1
+    && !await workspace.isEmpty(true)
+    && !await envFolderExists(workspaceDir, envNames[0])
```
Comment (adamro) [inline packages/cli/src/commands/env.ts:50]: env folder existing is enough for us?

Comment (netama) [reply packages/cli/src/commands/env.ts:50]: for deciding not to recommend? I think so (see the discussion in the comments on the ticket), because the cleanup logic guarantees the folder only exists if there is some file under it (at least as far as I understand), and we're assuming if the user moved something to env-specific then they intentionally left the rest under common. but happy to discuss more if you think that's not a good approach

File: packages/cli/src/prompts.ts (hunk 4 above)
Comment (netama) [reply packages/cli/src/prompts.ts:162]: @tomermevorach wdyt?

File: packages/cli/src/commands/env.ts (hunk 3 above)
Comment (netama) [reply packages/cli/src/commands/env.ts:202]: Discussed IRL and on the ticket - will add a generic `-y` flag ("accept all salto recommendations") in a separate PR

File: packages/cli/src/prompts.ts (hunk 4 above)
Comment (netama) [reply packages/cli/src/prompts.ts:162]: Will discuss tomorrow and then modify this copy in a separate PR

File: packages/cli/src/prompts.ts (hunk 5 above)
Comment (netama) [reply packages/cli/src/prompts.ts:173]: Will also discuss tomorrow - @tomermevorach fyi
